src/components/settings/SettingsSSOTab/steps/SSOConfigIDPStep.jsx

```diff
@@ -1,19 +1,42 @@
 import { Form } from '@edx/paragon';
+import { useContext, useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 import { useIdpState } from '../hooks';
+import { SSOConfigContext } from '../SSOConfigContext';
+import LmsApiService from '../../../../data/services/LmsApiService';
 
 /**
  * This step will opt user to enter one of these options:
  * 1: metadata url for IDP. In this case we want to read the metadata from the file directly.
  *    this is done in the backend. However, in this case, we don't yet know the entityId unless we read the file
  *    therefore, we want the user to also enter their entityID
  */
-const SSOConfigIDPStep = () => {
+const SSOConfigIDPStep = ({ enterpriseId }) => {
   const {
-    metadataURL, entryType, entityID,
+    metadataURL, entryType, entityID, publicKey, ssoUrl,
     handleMetadataURLUpdate, handleMetadataEntryTypeUpdate, handleEntityIDUpdate,
+    handlePublicKeyUpdate, handleSsoUrlUpdate,
   } = useIdpState();
 
-  const TITLE = 'First, select the method to provide your Identity Provider Metadata and fill out the corresponding fields. ';
+  const { setCurrentError, ssoState } = useContext(SSOConfigContext);
+  const { currentStep } = ssoState;
+
+  useEffect(() => {
```

Comment:
so... found out that previous iterations of this weren't actually fetching the provider data, just using the saml provider config fields... 🤦

Reply:
I'm now fetching the provider data on load and populating the fields with the data